Heroku example for the Deployment Guide

trevorturk opened the pull request:
Re: https://github.com/socketry/falcon-rails-example/issues/1, adding some documentation to show a full working example Heroku configuration for Falcon. 

Note I wasn't sure where to put this, but I believe this could be a reasonable place. I also tried to follow the new configuration style in the guides, but feedback is welcome. 

Note also that I included preload, cache, and count config examples which may or may not be desirable. I'm happy to adjust. We could also adjust to use a different ENV var etc as opposed to `FALCON_COUNT` -- feel free to adjust as you like, or let me know what you'd prefer and I'll make adjustments. 

Fixes https://github.com/socketry/falcon/issues/189
Fixes https://github.com/socketry/falcon/issues/121

## Types of Changes

- Documentation.

## Contribution

- [x] I agree to the [Developer's Certificate of Origin 1.1](https://developercertificate.org/).

ioquatix commented:
This looks great to me.

For all the major configuration options you've listed, it would be great to explain what it does, and how to use it.

trevorturk commented:
Ok, I made some updates. Let me know what you think! I added some more info and links. I changed `FALCON_COUNT` to `WEB_CONCURRENCY` since this is what most people use for Puma, but let me know what you think. I also removed `cache false` because from my testing it appears that the cache is disabled by default now. (Is that right? I thought I remembered it being enabled by default, which I found surprising, but perhaps we disabled it by default now?)

ioquatix commented:
Thanks, this looks great!
